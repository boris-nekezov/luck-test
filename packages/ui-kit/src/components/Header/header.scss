@import '../../styles/app.scss';

.header {
	@include theme-switcher() {
		border-bottom: $space-sm solid $active-primary-color--light;
	}

	display: flex;
	align-items: center;
	justify-content: space-between;
	padding: $space-md;
	margin-bottom: $space-lg;
	border-bottom: $space-xs solid $active-primary-color--dark;

	.logo {
		display: flex;
		align-items: center;
		padding: $space-md;

		&__image {
			width: 60px;
			height: 60px;
			margin-right: $space-sm;
			svg {
				position: relative;
				top: 14px;
				transform: scale(1.8);
				transform-origin: center;
			}
		}

		&__text {
			&_title {
				display: block;
				font-size: $base-font-size-md;
			}
			&_slogan {
				display: block;
				font-size: $base-font-size-sm;
			}
		}
	}
}
