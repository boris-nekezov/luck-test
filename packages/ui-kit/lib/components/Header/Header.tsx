import React from 'react';

import { Button } from '../Button/Button';
import ColorsIcon from '../../assets/colors.svg';

import './header.scss';

type User = {
  name: string;
};

export interface HeaderProps {
  user?: User;
  onLogin: () => void;
  onLogout: () => void;
  onCreateAccount: () => void;
}

export const Header = ({ user, onLogin, onLogout, onCreateAccount }: HeaderProps) => (
  <header>
    <div className="wrapper">
      <div>
        <ColorsIcon />
        <h1>Acme</h1>
      </div>
      <div>
        {user ? (
          <>
            <span className="welcome">
              Welcome, <b>{user.name}</b>!
            </span>
            <Button size="small" onClick={onLogout} label="Log out" />
          </>
        ) : (
          <>
            <Button size="small" onClick={onLogin} label="Log in" />
            <Button primary size="small" onClick={onCreateAccount} label="Sign up" />
          </>
        )}
      </div>
    </div>
  </header>
);
