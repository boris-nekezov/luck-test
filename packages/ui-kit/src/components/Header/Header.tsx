import Logo from '../../assets/logo.svg'

import './header.scss';

export interface HeaderProps {
  children?: JSX.Element | JSX.Element[],
}

export const Header = ({ children }: HeaderProps) => (
  <div className='header'>
    <div className="logo">
      <div className="logo__image">
        <Logo />
      </div>
      <div className="logo__text">
        <span className="logo__text_title">LUCK TEST</span>
        <span className="logo__text_slogan">CHECK YOU LUCK</span>
      </div>
    </div>
    {children}
  </div>
)

