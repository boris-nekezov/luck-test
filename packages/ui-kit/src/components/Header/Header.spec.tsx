import { render } from '@testing-library/react';
import { Header } from './Header';

describe('Given `Header` component', () => {
  let wrapper: ReturnType<typeof render>;

  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render(<Header />);
    });

    it('should match snapshot', () => {
      expect(wrapper.baseElement).toMatchSnapshot();
    })
  })
});
