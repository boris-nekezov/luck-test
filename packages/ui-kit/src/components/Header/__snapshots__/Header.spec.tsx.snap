// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given \`Header\` component when it is rendered should match snapshot 1`] = `
<body>
  <div>
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <div
          class="logo__image"
        >
          <logo.svg />
        </div>
        <div
          class="logo__text"
        >
          <span
            class="logo__text_title"
          >
            LUCK TEST
          </span>
          <span
            class="logo__text_slogan"
          >
            CHECK YOU LUCK
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
